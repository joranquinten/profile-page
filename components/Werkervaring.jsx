import React from 'react';
import renderHTML from 'react-render-html';
import styled from 'styled-components';
import { Timeline, Card } from 'antd';
import { colors } from '../styled/themes/default';

import { sorterByProperty } from '../common/helpers';

import { experience } from '../data/nl';

const StyledTimeline = styled.div`
  .ant-card {
    border-color: ${colors.blackDarkened};
    background: ${colors.blackLightened};
    color: ${colors.white};

    .ant-card-head {
      border-color: ${colors.blackDarkened};
    }

    .ant-card-head-title {
      color: ${colors.tertiaryColor};
    }
  }
`;

const Werkervaring = () => (
  <StyledTimeline>
    <Timeline>
      {experience
        .sort(sorterByProperty('tot', 'date'))
        .reverse()
        .map(item => (
          <Timeline.Item key={item.name} color={colors.primaryColor}>
            <Card title={item.name}>
              {`${item.van} - ${item.tot || 'heden'}`}
              <br />
              {renderHTML(item.description)}
            </Card>
          </Timeline.Item>
        ))}
    </Timeline>
  </StyledTimeline>
);

export default Werkervaring;
