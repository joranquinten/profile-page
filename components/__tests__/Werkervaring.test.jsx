import { shallow } from 'enzyme';

import Werkervaring from '../Werkervaring';

jest.mock('../../data/nl.js');

describe('<Werkervaring />', () => {
  it('should render without crashing', () => {
    expect(shallow(<Werkervaring />)).toMatchSnapshot();
  });
});
