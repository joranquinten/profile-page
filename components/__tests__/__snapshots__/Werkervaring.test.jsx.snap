// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Werkervaring /> should render without crashing 1`] = `
<Werkervaring__StyledTimeline>
  <Timeline
    mode=""
    reverse={false}
  >
    <TimelineItem
      color="#54D17E"
      key="Employer one"
      pending={false}
    >
      <Card
        title="Employer one"
      >
        2018-01-01 - heden
        <br />
        <p
          key="0"
        >
          Employer one description
        </p>
      </Card>
    </TimelineItem>
    <TimelineItem
      color="#54D17E"
      key="Employer two"
      pending={false}
    >
      <Card
        title="Employer two"
      >
        2017-01-01 - 2018-12-31
        <br />
        <p
          key="0"
        >
          Employer two description
        </p>
      </Card>
    </TimelineItem>
  </Timeline>
</Werkervaring__StyledTimeline>
`;
